service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantAboutToStartControllerTest.java

```diff
@@ -26,8 +56,94 @@ void shouldPrepopulateApplicantDataWhenNoApplicantExists() {
+
         AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
 
         assertThat(callbackResponse.getData()).containsKey("applicants");
     }
+
+    @BeforeEach
+    void setup() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+    }
+
+    @Test
+    void shouldAddOrganisationDetailsToApplicantWhenOrganisationExists() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("data", "some data"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+
+        CaseData data = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(data.getAllApplicants()).contains(buildApplicant());
+    }
+
+    @Test
+    void shouldFindOrganisation() {
+        Organisation organisation = buildOrganisation();
+        given(authTokenGenerator.generate()).willReturn(serviceAuthToken);
+        given(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).willReturn(organisation);
+
+        Organisation actualOrganisation = organisationApi.findOrganisationById(userAuthToken, serviceAuthToken);
+
+        assertThat(actualOrganisation).isEqualTo(organisation);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenOrganisationNotFound() {
+        Exception exception = new FeignException.NotFound("", REQUEST, new byte[]{});
+        when(organisationApi.findOrganisationById(userAuthToken, serviceAuthToken)).thenThrow(exception);
+
+        assertThatThrownBy(() -> organisationApi.findOrganisationById(userAuthToken, serviceAuthToken))
+            .isInstanceOf(FeignException.class);
+
+    }
+
+    private Organisation buildOrganisation() {
+        return Organisation.builder().name("Organisation")
+        .contactInformation(buildOrganisationContactInformation())
+            .build();
+    }
+
+    private ArrayList<ContactInformation> buildOrganisationContactInformation() {
+        ArrayList<ContactInformation> contactInformation = new ArrayList<>();
+        contactInformation.add(ContactInformation.builder()
+            .addressLine1("Flat 12, Pinnacle Apartments")
+            .addressLine1("Saffron Central")
+            .county("London")
+            .country("United Kingdom")
+            .postCode("CR0 2GE")
+            .build());
+
+        return  contactInformation;
+    }
+
+    private Element<Applicant> buildApplicant() {
+        return Element.<Applicant>builder()
+            .value(Applicant.builder()
+                .party(ApplicantParty.builder()
+            .organisationName("Organisation")
+                    .address(buildApplicantContactInformation())
+                    .build())
+                .build())
+            .build();
```

Comment:
you can use ElementUtils.element(Applicant.builder()...) static method here. Static import

Reply:
This method isn't needed anymore so i've removed